include/audio/dmic.h

```diff
@@ -0,0 +1,301 @@
+ *   * Neither the name of the Intel Corporation nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author:	Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+ *		Sathish Kuttan <sathish.k.kuttan@.intel.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef __AUDIO_DMIC_H__
+#define __AUDIO_DMIC_H__
+
+#include <kernel.h>
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * DMIC driver states
+ */
+enum dmic_state {
+	DMIC_STATE_UNINIT,	/* Uninitialized */
+	DMIC_STATE_INITIALIZED,	/* Initialized */
+	DMIC_STATE_CONFIGURED,	/* Configured */
+	DMIC_STATE_ACTIVE,	/* Active */
+	DMIC_STATE_PAUSED,	/* Paused */
+};
+
+/**
+ * DMIC driver trigger commands
+ */
+enum dmic_trigger {
+	DMIC_TRIGGER_STOP,	/* stop stream */
+	DMIC_TRIGGER_START,	/* start stream */
+	DMIC_TRIGGER_PAUSE,	/* pause the stream */
+	DMIC_TRIGGER_RELEASE,	/* release paused stream */
+	DMIC_TRIGGER_RESET,	/* reset */
+};
+
+/**
+ * PDM Channels LEFT / RIGHT
+ */
+enum pdm_lr {
+	PDM_CHAN_LEFT,
+	PDM_CHAN_RIGHT,
+};
+
+/**
+ * PDM Input/Output signal configuration
+ */
+struct pdm_io_cfg {
+	/* parameters global to all PDM controllers */
+
+	/* minimum clock frequency supported by the mic */
+	u32_t	min_pdm_clk_freq;
+	/* maximum clock frequency supported by the mic */
+	u32_t	max_pdm_clk_freq;
+	/* minimum duty cycle in % supported by the mic */
+	u8_t	min_pdm_clk_dc;
+	/* maximum duty cycle in % supported by the mic */
+	u8_t	max_pdm_clk_dc;
+
+	/* parameters unique to each PDM controller */
+
+	/* Bit mask to optionally invert PDM clock */
+	u8_t	pdm_clk_pol;
+	/* Bit mask to optionally invert mic data */
+	u8_t	pdm_data_pol;
+	/* Collection of clock skew values for each PDM port */
+	u32_t	pdm_clk_skew;
+};
+
+/**
+ * Configuration of the PCM streams to be output by the PDM hardware
+ */
+struct pcm_stream_cfg {
+	/*
+	 * if either rate or width is set to 0 for a stream,
+	 * the stream would be disabled
+	 */
+
+	/* PCM sample rate of stream */
+	u32_t			pcm_rate;
+	/* PCM sample width of stream */
+	u8_t			pcm_width;
+	u8_t			reserved;
```

Comment:
The reserved field is not needed.

Reply:
It was added for alignment. If convention is to not add such fields, I can remove it.